app/models/user.rb

```diff
@@ -1586,6 +1586,10 @@ def disable_redesigned_user_menu
     Discourse.redis.del("#{REDESIGN_USER_MENU_REDIS_KEY_PREFIX}#{self.id}")
   end
 
+  def no_likes_notifications?
+    user_option&.like_notification_frequency == UserOption.like_notification_frequency_type[:never]
```

Comment:
The logic in this method doesn't really belong in the `User` model since it is a `UserOption` concern. We can either move this method to `UserOption` or add a method delegate here.